src/test/java/org/broadinstitute/hellbender/engine/ReadsDataSourceUnitTest.java

```diff
@@ -587,6 +610,63 @@ public void testManuallySpecifiedIndicesWithCustomReaderFactory( final List<Path
     }
 
     @Test(dataProvider = "manuallySpecifiedIndexTestData")
+    public void testManuallySpecifiedIndicesWithCustomReaderFactoryAndNullWrappers( final List<Path> bams, final List<Path> indices ) {
+        final SamReaderFactory customFactory = SamReaderFactory.makeDefault().validationStringency(ValidationStringency.STRICT);
+        // ReadsDataSource should not be using the wrapper since the files are not on the Google cloud.
+        // So we pass this invalid wrapper: if the code tries to use it, it'll blow up.
+        Function<SeekableByteChannel, SeekableByteChannel> nullWrapper = (SeekableByteChannel) -> null;
+
+        try ( final ReadsDataSource readsSource = new ReadsDataSource(bams, indices, customFactory, nullWrapper, nullWrapper) ) {
+            Assert.assertTrue(readsSource.indicesAvailable(), "Explicitly-provided indices not detected for bams: " + bams);
+
+            final Iterator<GATKRead> queryReads = readsSource.query(new SimpleInterval("1", 1, 300));
+            int queryCount = 0;
+            while ( queryReads.hasNext() ) {
+                ++queryCount;
+                queryReads.next();
+            }
+            Assert.assertEquals(queryCount, 5, "Wrong number of reads returned in query");
+        }
+    }
+
+    @Test(dataProvider = "cloudXorTestData", groups={"bucket"})
+    public void testCloudBamWithCustomReaderFactoryAndWrappers( final List<Path> bams, final List<Path> indices ) {
+        final SamReaderFactory customFactory = SamReaderFactory.makeDefault().validationStringency(ValidationStringency.STRICT);
+        // The input files are XOR'd with a constant. We use a wrapper to XOR it back.
+        // If the code uses the wrong wrapper, or omits one, then the test will fail.
+        Function<SeekableByteChannel, SeekableByteChannel> xorData = XorWrapper.forKey((byte)74);
+        Function<SeekableByteChannel, SeekableByteChannel> xorIndex = XorWrapper.forKey((byte)80);
+
+        try ( final ReadsDataSource readsSource = new ReadsDataSource(bams, indices, customFactory, xorData, xorIndex) ) {
+            Assert.assertTrue(readsSource.indicesAvailable(), "Explicitly-provided indices not detected for bams: " + bams);
+
+            final Iterator<GATKRead> queryReads = readsSource.query(new SimpleInterval("1", 1, 300));
+            int queryCount = 0;
+            while ( queryReads.hasNext() ) {
+                ++queryCount;
+                queryReads.next();
+            }
+            Assert.assertEquals(queryCount, 2, "Wrong number of reads returned in query");
+        }
+    }
+
+    // this code generates the xor'd inputs, in case we need to do it again.
+    private void createXorInput() throws IOException {
+        String prefix = "src/test/resources/org/broadinstitute/hellbender/engine/";
+        Path input = Paths.get(prefix + "reads_data_source_test2.bam");
+        Path output = Paths.get(prefix + "reads_data_source_test4.xor.bam");
```

Comment:
Can you write the output to a temporary file (as returned by the inherited method `createTempFile()`), instead of writing to a file in `src/test/resources`?